### PR TITLE
[feat] better clarify invalid named layout references

### LINKED ISSUE #5903: Unclear error when a layout endpoint has a named layout inheritance in their filename.
### Describe the bug

Adding an inherited layout naming convention to a layout endpoint throws an unclear error during the sveltekit sync process.

Instead, the thrown error should tell the user to rename their file from `layout@example.server.ts` to `layout.server.ts`.

This happened to me while I was still in the mindset of naming the endpoint files exactly the same as the page file before the new breaking changes.

I could not find any clarification about this in the official docs about this. Could it also be helpful to migrating users by clarifying this in the docs?

An example of the correct filenames would be:
```
+layout-root.svelte
+layout@root.svelte
+layout.ts
+layout-another@root.svelte
+layout-another.ts
```

### Reproduction

https://github.com/s3812497/sveltekit-v408-bug

### Logs

```shell
❯ pnpm run dev

> my-app@0.0.1 dev /Users/teeming/my-app
> vite dev

error when starting dev server:
Error: Files and directories prefixed with + are reserved (saw src/routes/+layout@root.ts)
    at file:///Users/teeming/my-app/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.408_svelte@3.49.0+vite@3.0.7/node_modules/@sveltejs/kit/src/core/sync/create_manifest_data/index.js:73:11
    at Array.forEach (<anonymous>)
    at create_manifest_data (file:///Users/teeming/my-app/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.408_svelte@3.49.0+vite@3.0.7/node_modules/@sveltejs/kit/src/core/sync/create_manifest_data/index.js:60:39)
    at Module.create (file:///Users/teeming/my-app/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.408_svelte@3.49.0+vite@3.0.7/node_modules/@sveltejs/kit/src/core/sync/sync.js:25:24)
    at update_manifest (file:///Users/teeming/my-app/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.408_svelte@3.49.0+vite@3.0.7/node_modules/@sveltejs/kit/src/vite/dev/index.js:58:35)
    at dev (file:///Users/teeming/my-app/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.408_svelte@3.49.0+vite@3.0.7/node_modules/@sveltejs/kit/src/vite/dev/index.js:214:8)
    at async Object.configureServer (file:///Users/teeming/my-app/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.408_svelte@3.49.0+vite@3.0.7/node_modules/@sveltejs/kit/src/vite/index.js:459:11)
    at async createServer (file:///Users/teeming/my-app/node_modules/.pnpm/vite@3.0.7/node_modules/vite/dist/node/chunks/dep-0f13c890.js:59303:28)
    at async CAC.<anonymous> (file:///Users/teeming/my-app/node_modules/.pnpm/vite@3.0.7/node_modules/vite/dist/node/cli.js:699:24)
 ELIFECYCLE  Command failed with exit code 1.
```

### System Info

```shell
System:
    OS: macOS 12.5
    CPU: (8) arm64 Apple M1 Pro
    Memory: 2.75 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
  Browsers:
    Brave Browser: 104.1.42.95
    Chrome: 104.0.5112.79
    Firefox: 103.0.1
    Firefox Developer Edition: 103.0
    Safari: 15.6
    Safari Technology Preview: 16.0
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.64 
    @sveltejs/kit: next => 1.0.0-next.408 
    svelte: ^3.44.0 => 3.49.0 
    vite: ^3.0.0 => 3.0.7
```

### Severity

annoyance

### Additional Information

confusing for the developer

### PR DESCRIPTION
**Pull request by dummdidumm**
Closes #5903

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: documentation/docs/04-advanced-routing.md [hunk 1]
```diff
@@ -144,6 +144,8 @@ Some parts of your app might need something other than the default layout. For t
 <h1>I am inside +layout-foo</h1>
 ```
 
+> Only the Svelte file should reference the named layout, not its accompanying `+page.js` for example
```
Comment (ignatiusmb) [inline documentation/docs/04-advanced-routing.md:147]: How about this?

```suggestion
> Named layout should only be referenced from the `+page.svelte` file
```

Comment (dummdidumm) [reply documentation/docs/04-advanced-routing.md:147]: Oh yeah that's way better 👍

**Review by ignatiusmb — APPROVED** [review]
Looks good! The errors and test still has the previous sentence, btw.

File: documentation/docs/04-advanced-routing.md [hunk 2]
```diff
@@ -144,6 +144,8 @@ Some parts of your app might need something other than the default layout. For t
 <h1>I am inside +layout-foo</h1>
 ```
 
+> Named layout should only be referenced from `+page.svelte` files
```
Comment (dummdidumm) [inline documentation/docs/04-advanced-routing.md:147]: Just realized this isn't enough, it also applies to (named) layout files. I prefer to not be too wordy, here so I'll change this to
```suggestion
> Named layout should only be referenced from Svelte files
```
